tools/driver/CMakeLists.txt

```diff
@@ -38,6 +38,11 @@ if(SWIFT_HOST_VARIANT MATCHES "${SWIFT_DARWIN_VARIANTS}")
       LINK_FLAGS " -lobjc -Wl,-force_load,\"${SWIFT_REPL_ARCLITE}\"")
 endif()
 
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
+  set_property(TARGET swift APPEND_STRING PROPERTY
+      LINK_FLAGS " -Wl,--allow-multiple-definition")
```

Comment:
I'll remove this part. Updating clang to 3.8.1 with recent released cygwin package, the link flag can be removed.

With some test code, I found the Cygwin clang compiler 3.7.1 generates wrong section type for the data section which has a guard variable for template-static-member. That leads to linker's multiple definition message.